pghoard/config.py

```diff
@@ -23,16 +22,36 @@ def get_cpu_count():
     return multiprocessing.cpu_count()
 
 
-def find_pg_binary(program, versions=None):
+def find_pg_binary(programs, versions=None):
+    if programs == 'pg_receivexlog':
+        programs = ['pg_receivexlog', 'pg_receivewal']
+    if isinstance(programs, str):
+        programs = [programs]
     pathformats = ["/usr/pgsql-{ver}/bin/{prog}", "/usr/lib/postgresql/{ver}/bin/{prog}"]
-    for ver in versions or SUPPORTED_VERSIONS:
+    versions = versions or SUPPORTED_VERSIONS
+    for ver in versions:
         for pathfmt in pathformats:
-            if LooseVersion(ver) >= "10" and program == "pg_receivexlog":
-                program = "pg_receivewal"
-            pgbin = pathfmt.format(ver=ver, prog=program)
-            if os.path.exists(pgbin):
-                return pgbin, ver
-    return os.path.join("/usr/bin", program), None
+            for program in programs:
+                pgbin = pathfmt.format(ver=ver, prog=program)
+                if os.path.exists(pgbin):
+                    return pgbin, ver
+    # We couldn't find a supported version in the "well known locations",
+    # let's search in the path.
+    for path in os.environ["PATH"].split(os.pathsep):
+        for program in programs:
+            command = os.path.join(path, program)
+            print("Looking at %s" % command)
+            if os.path.exists(command) and os.access(command, os.X_OK):
+                try:
+                    version_output = subprocess.check_output([command, "--version"])
+                    version_string = version_output.decode("ascii").strip()
+                    version_int = convert_pg_command_version_to_number(version_string)
+                    version_string = str(int(version_int / 10000))
+                    if version_string in versions:
+                        return command, version_string
+                except Exception as _:
+                    pass
+    return None, None
```

Comment:
I think this should be changed to a raise RuntimeError("Cannot find program(s): %s".format(", ".join(programs_to_search_for))

Reply:
Done.